packages/icons/src/library/caution.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * WordPress dependencies
+ */
+import { SVG, Path } from '@wordpress/primitives';
+
+const caution = (
```

Comment:
@WordPress/gutenberg-design I just want to make sure we measure twice and cut once here because publishing a new icon is kind of irreversible 😄

Would it be worth renaming these so it follows the convention of `foo` + `fooFilled` like we do for the other pairs like this? For example, change the canonical name of `warning` to `cautionFilled` (maintaining back compat with an aliased export of course).

<img width="219" alt="Caution and warning icons" src="https://github.com/user-attachments/assets/9a74491f-97ff-4a3e-997b-3a1482a2295a" />

<img width="227" alt="Help icon pair" src="https://github.com/user-attachments/assets/8ed1e9f2-611d-41bb-8647-39b0c6bf3392" />

Reply:
Oh, yes, good suggestion. Just to clarify; that means any existing instances of `warning` would map to `cautionFilled`?

We should update the `cautionFilled` icon so that the details match the new `caution` icon too. Do you think it makes sense to do that in this PR, or separately?

Reply:
> that means any existing instances of `warning` would map to `cautionFilled`?

Correct 👍 

> Do you think it makes sense to do that in this PR, or separately?

Yes, let me pull the icon changes out of this PR so we can address it separately with proper care. I'll prepare a PR so you can push up changes to the SVG.

@Vrishabhsk Keeping this PR on hold for just a bit more. Thank you for your patience!

Reply:
@jameskoster Updates to the SVG can be pushed here: #67895